src/plugins/auto/auto_schedule.cpp

```diff
@@ -79,25 +79,75 @@ void AutoSchedule::GenerateWorkers(const std::string& device,
                 IdleGuard<NotBusyPriorityWorkerRequests> idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
                 workerRequestPtr->_exceptionPtr = exceptionPtr;
                 {
-                    auto capturedTask = std::move(workerRequestPtr->_task);
-                    capturedTask();
-                }
-                // try to return the request to the idle list (fails if the overall object destruction has began)
-                if (idleGuard.Release()->try_push(std::make_pair(workerRequestPtr->_index, workerRequestPtr))) {
-                    // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                    // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                    IE::Task t;
-                    do {
-                        _inferPipelineTasks.try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t)));
-                    do {
-                        _inferPipelineTasksDeviceSpecific[device]->try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t), device));
+                    if (workerRequestPtr->_exceptionPtr != nullptr) {
+                        if (workerRequestPtr->_testExec->count == 1) {
+                            auto capturedTask = std::move(workerRequestPtr->_task);
```

Comment:
does this flag means we only try once after failure? I think we don't need to pose this restriction here in infer request?

Reply:
Yes, if there is no other devices in _devicePriorities and can retry in many times, it will hang here. I think it' better to retry once after selectOtherDevice.

Reply:
Added the result of `selectOtherDevice()`, so rm the count flag now.